### DISCUSSION TITLE
Substructures not shown in Streamlit

### DISCUSSION THREAD

**Discussion by napoles-uach** [rdkit/rdkit#4779] [Q&A]
Hi there! I'm trying to do this [example](https://gist.github.com/greglandrum/5d45b56afe75603b955103cdd0d8e038) in Streamlit. Here is the code:
```
import streamlit as st
from rdkit import Chem
from rdkit.Chem import Draw
from PIL import Image

compound_smiles = 'CN1C=NC2=C1C(=O)N(C(=O)N2C)C'
m = Chem.MolFromSmiles(compound_smiles)
im=Draw.MolToImage(m)

m = Chem.MolFromSmiles('c1cc(C(=O)O)c(OC(=O)C)cc1')
substructure = Chem.MolFromSmarts('C(=O)O')
m.GetSubstructMatches(substructure)
im=Draw.MolToImage(m)
st.image(im)
```

This shows the image of the molecule, but not the highlighted substructure. Does anyone have an idea of how to solve this??
Thanks!!

**Comment by greglandrum** [comment]
@napoles-uach : the automatic highlighting in that example is something which only takes place in jupyter.
The documentation includes an example of how to generate drawings with highlighting:
https://www.rdkit.org/docs/GettingStartedInPython.html#drawing-molecules
Note that all of those examples don't show the final two steps to get the image data from the `MolDraw2D` object:
```
d.FinishDrawing()
data = d.GetDrawingText()
```
If you're using a `MolDraw2DSVG` object that last line returns the SVG text. If you're using a `MolDraw2DCairo` object it returns a binary string with PNG data.
